Preprocessing/tst_data.py

```diff
@@ -0,0 +1,1357 @@
+"""
+Common functions for preprocessing
+"""
+
+import os
+import re
+import glob
+import json
+import copy
+import argparse
+import numpy as np
+import pandas as pd
+
+EXPERIMENTS_FOLDER = os.path.abspath(f"{__file__}/../../Data")
+RAW_EXPERIMENT_FP_FOLDERS = sorted(
+    filter(lambda fp: os.path.isdir(fp), glob.glob(f"{EXPERIMENTS_FOLDER}/raw/TST_*"))
+)
+
+
+class NotAnExperimentFolder(Exception):
```

Comment:
@sbancal This file is enormous. I could not find the time to review it. Maybe it was not possible to split it in different files. And maybe we should have something more than the pep8 for those kind of things. But clearly. It's not possible to review this kind of file in less than 5mn.

Large files tend to do a lot of things and can make it hard following what's going. While there is not an objective maximum number of lines considered acceptable in a file, most people would agree it should not be in the thousands. Recommendations usually range from 100 to 500 lines.

Same for function/classes (usually less than 20. too big when >50 for sure).  Usually not more than one level of indentation. https://stackoverflow.com/questions/611304/how-many-lines-of-code-should-a-function-procedure-method-have good references here.

I'm not saying you should rewrite that file for this pull request. It's clearly too late for that. I'm just saying that it failed to be easily reviewable. I don't know how @charlottegiseleweil approached it. But I could not find the mental space to take in that many lines of code. Hope you find this comment helpful. That's. my intention at least. I'd be glad if we could find time to discuss that further when you have the time

Reply:
I understand ... and agree that such code review is almost impossible with few time.

About the file length, I realize that source code reaching 1K lines happen regularly in my projects.
Good topic to discuss on occasion.

Thanks